website/blog/2024-03-18-deep-learning-part-2/index.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Fine Tuning LLMs with MLflow: Deep Learning with MLflow (Part 2)
+description: Highlighting the recent improvements in MLflow for Deep Learning workflows
+slug: deep-learning-part-1
+authors: [abe-omorogbe, avinash-sooriyarachchi, puneet-jain]
+tags: [Deep Learning]
+thumbnail: img/blog/dl-chart-grouping.gif
+---
+
+![](deep-learning-pt2-min.jpg)
+
+In the realm of deep learning, finetuning of pre-trained Large Language Models (LLMs) on private datasets stands as a critical application. This practice is not only common but also essential for developing specialized models, particularly for tasks like custom Named Entity Recognition (NER). In such scenarios, tools like MLflow are invaluable. They ensure that every aspect of the training process - metrics, parameters, and artifacts - are reproducibly tracked, logged, analyzed, compared, and shared. In this tutorial we are going to be using [MLflow 2.11](https://mlflow.org/releases/2.11.0) and [the new MLflow Deep Learning features](https://mlflow.org/blog/deep-learning-part-1) to track all the important aspects of this project.
+
+
+# Use Case: Fine Tuning Llama2-7B for NER 
+
+A prime example of this application is the finetuning of an Llama2-7B pre-trained model for instruction following. The goal is to extract structured information from unstructured text, a task central to NER.  Selecting the Llama2 7B model is a strategic decision. Its size strikes a perfect balance, having a large enough parameter count to handle intricate tasks while remaining compact enough for fine tuning on a modest cluster of less powerful GPUs like Nvidia T4s, thanks to Parameter Efficient Fine Tuning (PEFT) methods. Additionally, it can be finetuned on a single A100 GPU utilizing PEFT techniques quite easily.
```

Comment:
Recommend clarifying the task & providing a clear example, in the text, of input and expected output. It's not clear here whether it's instruction following, NER, or structured information extraction in a way that is central to, but still distinct from, NER.